TF_Tooling_Web/Plugins/SnappingPlugin/src/Ray.ts

```diff
@@ -88,8 +88,11 @@ export class Ray {
         return pos;
     }
 
-    public static Hit(position: Vector2, element: Element): boolean {
-        let elementsAtPos: Element[] = document.elementsFromPoint(position.x, position.y);
-        return elementsAtPos.find((value: Element, index: number, obj: Element[]) => value === element) !== undefined;
+    public static Hit(position: Vector2, rect: DOMRect): boolean {
+
+        return !(rect.right < position.x ||
```

Comment:
Either left/right or top/bottom logic looks reversed though the result does appear to behave as expected.

Might be easier to read as using AND logic rather than NOT OR as it is now.

I expect a hit to be:
```
position.x > rect.left && 
position.x < rect.right && 
position.y > rect.bottom && 
position.y < rect.top
```

Reply:
The logic error was down to me not realising the y axis origin is at the top of the page.